### ISSUE TITLE
JDBC connectors will not clean up temp tables if write fails

### ISSUE THREAD

**Issue by tdcmeehan** [prestodb/presto#11576]
When performing a write, Presto will create a temp table, write to that temp table, and then copy the rows from the temp table to the destination table in one shot at the end.

I have found that if any exception occurs during a write, then Presto won't attempt to drop the temp table it was writing to, resulting in an unused temp table which may have data.

Here is a test case in `TestPostgresSqlIntegrationSmokeTest` which reproduces the problem:

```java
    @Test
    public void testInsertWithFailureDoesntLeaveBehindOrphanedTable()
            throws SQLException
    {
        assertUpdate("CREATE TABLE test_cleanup (" +
                "column_a INTEGER" +
                ")");

        assertQuery("SHOW TABLES", "VALUES 'orders', 'test_cleanup'");

        execute("ALTER TABLE tpch.test_cleanup ADD CHECK (column_a > 0)");

        Exception thrown = null;
        try {
            assertUpdate("INSERT INTO tpch.test_cleanup(column_a) VALUES (0)", 1);
        }
        catch (Exception e) {
            thrown = e;
            assertQuery("SHOW TABLES", "VALUES 'orders', 'test_cleanup'"); // Assertion fails due to temp table
        }
        assertNotNull(thrown, "Expected exception to be thrown when attempting to insert");
    }
```

This will fail on the second `SHOW TABLES` assertion:

```console
java.lang.AssertionError: For query: 
 SHOW TABLES
not equal
Actual rows (up to 100 of 1 extra rows shown, 3 matching and extra rows in total):
    [tmp_presto_db4cf516077a46399b8e1e66b60642a6]
Expected rows (up to 100 of 0 missing rows shown, 2 matching and missing rows in total):
```

**Comment by tdcmeehan** [comment]
Fixed in #11561, as this was blocking my ability to write some tests

**Comment by tdcmeehan** [comment]
Merged #11577
